### PR TITLE
add darkmode with styles

### PR DESCRIPTION
**Pull request by REFAEI91**

### CODE REVIEW

File: components/DarkMode/DarkMode.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      <Head>
+        <style>
+          {`
+            body.dark-mode {
+              background-color: #333;
+              color: #fff;
+            }
+
+            .sun {
+              filter:  sepia(1) saturate(1) hue-rotate(180deg);
```
Comment (ThomasKirst) [inline components/DarkMode/DarkMode.js]:
```suggestion
              filter: sepia(1) saturate(1) hue-rotate(180deg);
```

File: components/DarkMode/DarkMode.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+          onChange={toggleDarkMode}
+        />
+        <Label htmlFor="dark-mode-toggle">
+          <StyledImageSun
+            src="/sun.svg"
+            alt="sun"
+            width={25}
+            height={25}
+            className={isDarkMode ? "sun dark-mode" : "sun"}
+          />
```
Comment (ThomasKirst) [inline components/DarkMode/DarkMode.js]: We generally advise to stick to one variant of using styles, either the usage of styled components or the usage of classes. Mixing these both is not considered a consistent approach.
Could you make this work with passing a prop to the styled component and decide thereafter which css rules should apply based on the prop `isDarkMode`?

```js
export const StyledImageSun = styled(Image)`
  position: absolute;
  top: 1px;
  z-index: 100;
  filter: ${({ darkMode }) => (/* rules to apply */)};
  fill: ${({ darkMode }) => (darkMode ? /* rules to apply ... */)};
`;

<StyledImageSun
  src="/sun.svg"
  alt="sun"
  width={25}
  height={25}
  darkMode={isDarkMode}
/>
```
You could then avoid the styles in the `Head` area of your page.

The class for the `body` element could be moved to the global styles file `styles.js`.

**Review by ThomasKirst — CHANGES_REQUESTED** [review]
Hi Raffaei,
great work overall, looks nice :)
I just have one comment regarding the styling (styled components / global styles).
– Thomas

File: components/DarkMode/DarkMode.js (hunk 1 above)
Comment (REFAEI91) [reply components/DarkMode/DarkMode.js]: thank you , all I have already did the Suggested changes

File: components/DarkMode/DarkMode.js (hunk 2 above)
Comment (REFAEI91) [reply components/DarkMode/DarkMode.js]: Thank you for your feedback , i have already apply all the suggested Changes

File: styles.js [hunk 3]
```diff
@@ -11,16 +11,19 @@ export default createGlobalStyle`
...
-    font-family: ${kanit.style.fontFamily};
-    padding-bottom: 3rem;
+  margin: 0;
+  background-color: lightyellow orchid;
+  font-family: ${kanit.style.fontFamily};
+  padding-bottom: 3rem;
+  &.isDarkMode {
+  color: #fff;
+  background-color: #43474c;
+  }
```
Comment (ThomasKirst) [inline styles.js:27]: The formatting seems a bit off. Please check your prettier settings and format the document (CMD-SHIFT-P => Format Document).

**Review by ThomasKirst — APPROVED** [review]
Great how you adapted your code.
I just found a little formatting issue in your global styles file which you should fix, and then you are good to merge 😃

**Pull request merged**